subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/testng/TestNGOptions.java

```diff
@@ -249,6 +257,9 @@ public void setListeners(Set<String> listeners) {
      *
      * If not present, parallel mode will not be selected
      */
+    @Input
+    @Optional
+    @Nullable
```

Comment:
❌ Parallel is not an input, as it doesn't affect which tests are run or what the result is.

The same goes for thread count, listeners and similar properties that only affect the execution model, but not the result.